sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -17491,4 +17491,36 @@ public void testEmptyGroupWithOffsetDoesntInfiniteLoop() throws Exception
         ImmutableList.of()
     );
   }
+
+  @Test
+  public void testJoinWithTimeDimension() throws Exception
+  {
+    testQuery(
+        PLANNER_CONFIG_DEFAULT,
+        QUERY_CONTEXT_DONT_SKIP_EMPTY_BUCKETS,
```

Comment:
why not skipping empty buckets?

Reply:
I was facing some issues while setting up default context earlier, but now when I checked it again the default context works fine too. updated it with default context (using default context explicitly to avoid vectorization path)